pkgs/development/libraries/aws-sdk-cpp/gcc11.patch

```diff
@@ -0,0 +1,164 @@
+diff --git a/aws-cpp-sdk-cloudfront-integration-tests/CloudfrontOperationTest.cpp b/aws-cpp-sdk-cloudfront-integration-tests/CloudfrontOperationTest.cpp
```

Comment:
Where does this diff come from? If it corresponds to an upstream commit, I generally prefer using fetchpatch since it provides more context into the patches and keeps the nixpkgs tree lean.

Reply:
I made those changes but not interested in ustreaming / contributing to amazon / aws.

if someone wants to upstream them I release them to the extend possible to the public domain. (And it was only adding missing `#include <thread>` anyways so doubt it is copyrightable in most jurisdictions to begin with)

Reply:
Ah, gotcha! no pressure to upstream necessarily. If it's just adding `#include <thread>` in a few places is there a reason to prefer a patch file instead of a simple sed/substituteInPlace? Using sed or something similar tends to be more robust to version upgrades down the line.

Reply:
I have little to no experience with sed. Closest thing I have done are regexes but would need a stable reference across the files to substitute for

Reply:
I usually do stuff like
```
sed '1i#include <thread>' -i file1 file2 some/path/{file3,file34}.cc
```

Reply:
Will attempt rewriting like so

Reply:
@fabianhjr no pressure to use sed per se. I why is be highly unintuitive. But any string replacement type tool should do the trick!

Reply:
I usually keep inserting before the first line, so that I don't accidentally insert inside a multi-line comment etc. but if it works this way...

Reply:
Didn't work either with `1i` nor `4i`, some headers would be missing.

Reply:
Reset to patch from the sed attempt

Reply:
What headers would be missing? This should absolutely be feasible with sed

Reply:
<thread>

Tried sed with 1i and 4i as mentioned: https://github.com/NixOS/nixpkgs/commit/323f044be4abdf58db941ed5d22c7cf7f0735c03

Reply:
Yes, I understand but this is absolutely possible to do with sed. So there must be something awry with your usage.

Reply:
What was the error exactly? We may be able to help if provided more information.

Reply:
NVM, it worked with `5i`, seems like there is some weird formatting issue and it sometimes ended up in a comment.

Reply:
`sed '5i#include <thread>' -I \` works. Please update the PR.

Inserting on line 1 doesn't work because some of the files have byte order markers (BOM), which sed does not handle. Inserting on line 4 doesn't work because that's in the middle of a multi-line comment.

Reply:
ok looks like you discovered the same thing independently!

Reply:
Didn't know about BOMs :O

Reply:
OMG, people actually use them in source files?

Reply:
Apparently AWS does in their infinite wisdom... but only in some files!

Reply:
We might blame `sed` that it doesn't transparently support them.